make/autoconf/jdk-options.m4

```diff
@@ -438,6 +438,15 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
           # It's harmless to be suppressed in clang as well.
           ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
           ASAN_LDFLAGS="-fsanitize=address"
+          # detect_stack_use_after_return causes ASAN to offload stack-local
+          # variables to c-heap and therefore breaks assumptions in hotspot
+          # that rely on data (e.g. Marks) living in thread stacks.
+          if test "x$TOOLCHAIN_TYPE" = "xgcc"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS --param asan-use-after-return=0"
+          fi
+          if test "x$TOOLCHAIN_TYPE" = "xclang"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS -fsanitize-address-use-after-return=never"
+          fi
```

Comment:
There's no change being made for the microsoft toolchain.  It seems like the same issues with the fake stack
should arise there.

Reply:
So I have tried it in AWS and added the [new comment](https://github.com/openjdk/jdk/pull/19843/commits/08153d9cf2ab8d0721fc1f257f9d662de4814212) about it as MSVC has it off by default.
But I have filed [JDK-8335228](https://bugs.openjdk.org/browse/JDK-8335228) as the MS-Windows build crashes if one explicitly enables `-fsanitize-address-use-after-return` there.

Reply:
I guess we'll have to monitor this in future VS updates, which might start enabling it.  But this seems okay for now.